File: src/main/scala/com/scala/test/model/ProduitEnrichi.scala
package com.scala.test.model

import java.math.BigDecimal

import play.api.libs.json.{JsNumber, JsValue, Json, Writes}

case class ProduitEnrichi(
                           id: Option[Long],
                           name: Option[String],
                           price: Option[BigDecimal])

object ProduitEnrichi {
  implicit val bigdecimalWrites = new Writes[BigDecimal] {
    def writes(o: BigDecimal): JsValue = {
      JsNumber(o.setScale(2))
    }
  }
  implicit val format = Json.format[ProduitEnrichi]
}

File: kafka-stream/src/main/java/com/java/kafka/stream/model/ProduitEnrichi.java
package com.java.kafka.stream.model;

import java.math.BigDecimal;

public class ProduitEnrichi {

    private Long id;
    private String name;
    private BigDecimal price;

    public ProduitEnrichi(Long id, String name, BigDecimal price) {
        this.id = id;
        this.name = name;
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "ProduitEnrichi{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", price=" + price +
                '}';
    }
}
